iterator.go

```diff
@@ -0,0 +1,267 @@
+package lotusdb
+
+import (
+	"bytes"
+	"container/heap"
+
+	"github.com/dgraph-io/badger/v4/y"
+	"github.com/rosedblabs/wal"
+)
+
+// IteratorI
+type IteratorI interface {
```

Comment:
1. the name is not very proper, maybe `baseIterator`?
2. this interface is not invoked by the user, so it should be lowercase.